File: atcoder/abc171B_mix_juice.nim
# Vicfred
# https://atcoder.jp/contests/abc171/tasks/abc171_b
# sorting, greedy
import strutils, sequtils, algorithm

var n, k: int
(n, k) = stdin.readLine.split.map parseInt
var p = stdin.readLine.split.map parseInt

p.sort

var ans = 0
for i in 0..<k:
  ans += p[i]

echo ans

File: atcoder/abc171B_mix_juice.ml
(* uninhm
 * https://atcoder.jp/contests/abc171/tasks/abc171_b
 * sorting
 * *)

Printf.printf "%d\n" @@
Scanf.scanf "%d %d\n" @@ fun n k ->
    let a = Array.init n @@ fun _ -> Scanf.scanf "%d " @@ fun i -> i in
    Array.sort (-) a;
    let ans = ref 0 in
    for i = 0 to k-1 do
        ans := !ans + a.(i)
    done;
    !ans;;
